src/particles/pusher/BeamParticleAdvance.cpp

```diff
@@ -186,19 +209,75 @@ AdvanceBeamParticlesSlice (
+                        +  uz_intermediate*uz_intermediate)*inv_c2 );
+                    // Estimation of the velocity at intermediate time
+                    const amrex::ParticleReal vx_n = ux_intermediate*gamma_intermediate_inv
+                                                     *PhysConstSI::c*inv_clight;
+                    const amrex::ParticleReal vy_n = uy_intermediate*gamma_intermediate_inv
+                                                     *PhysConstSI::c*inv_clight;
+                    const amrex::ParticleReal vz_n = uz_intermediate*gamma_intermediate_inv
+                                                     *PhysConstSI::c*inv_clight;
+                    const amrex::ParticleReal bx_n = vx_n*inv_clight_SI;
+                    const amrex::ParticleReal by_n = vy_n*inv_clight_SI;
+                    const amrex::ParticleReal bz_n = vz_n*inv_clight_SI;
+
+                    // Lorentz force over charge
+                    const amrex::ParticleReal flx_q = (Exp + vy_n*Bzp - vz_n*Byp);
+                    const amrex::ParticleReal fly_q = (Eyp + vz_n*Bxp - vx_n*Bzp);
+                    const amrex::ParticleReal flz_q = (Ezp + vx_n*Byp - vy_n*Bxp);
+                    const amrex::ParticleReal fl_q2 = flx_q*flx_q + fly_q*fly_q + flz_q*flz_q;
+
+                    // Calculation of auxiliary quantities
+                    const amrex::ParticleReal bdotE = (bx_n*Exp + by_n*Eyp + bz_n*Ezp);
+                    const amrex::ParticleReal bdotE2 = bdotE*bdotE;
+                    const amrex::ParticleReal coeff = gamma_intermediate*gamma_intermediate*(fl_q2-bdotE2);
+
+                    //Compute the components of the RR force
+                    const amrex::ParticleReal frx =
+                        RRcoeff*(PhysConstSI::c*(fly_q*Bzp - flz_q*Byp) + bdotE*Exp - coeff*bx_n);
+                    const amrex::ParticleReal fry =
+                        RRcoeff*(PhysConstSI::c*(flz_q*Bxp - flx_q*Bzp) + bdotE*Eyp - coeff*by_n);
+                    const amrex::ParticleReal frz =
+                        RRcoeff*(PhysConstSI::c*(flx_q*Byp - fly_q*Bxp) + bdotE*Ezp - coeff*bz_n);
+
+                    //Update momentum using the RR force
+                    // in normalized units wp_inv normalizes the time step
+                    // *clight/inv_clight_SI converts to proper velocity
+                    ux_next += frx*dt*wp_inv*clight*inv_clight_SI;
+                    uy_next += fry*dt*wp_inv*clight*inv_clight_SI;
+                    uz_next += frz*dt*wp_inv*clight*inv_clight_SI;
+                }
+
                 /* computing next gamma value */
                 const amrex::ParticleReal gamma_next_inv = 1._rt / std::sqrt( 1._rt
-                    + ux_next*ux_next*clightsq
-                    + uy_next*uy_next*clightsq
-                    + uz_next*uz_next*clightsq );
+                    + (ux_next*ux_next
+                    +  uy_next*uy_next
+                    +  uz_next*uz_next)*inv_c2 );
```

Comment:
Again confusing indent
```suggestion
                    + ( ux_next*ux_next
                       +uy_next*uy_next
                       +uz_next*uz_next )*inv_c2 );
```